src/Tasks/Microsoft.NET.Build.Tasks/JoinItems.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Linq;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class JoinItems : TaskBase
```

Comment:
I don't think so. We have an atrocious pattern in msbuild xml elsewhere for this that we need to eradicate. A task is a step in the right direction for those cases but I still worry about the perf here introducing this complexity where it didn't already exist. Feels like a risky fix at best. 

Dave, can we measure the impact of this on the Roslyn scenarios you've been tuning before merging?